src/request.js

```diff
@@ -217,7 +217,9 @@ export function getNodeRequestOptions(request) {
 	}
 
 	// HTTP-network-or-cache fetch step 2.11
-	if (!headers.has('User-Agent')) {
+	if (headers.get('User-Agent') === 'null') {
+		headers.delete('User-Agent');
+	} else if (!headers.has('User-Agent') && headers.get('User-Agent') !== 'null') {
 		headers.set('User-Agent', 'node-fetch (+https://github.com/node-fetch/node-fetch)');
 	}
```

Comment:
@Richienb Is this commit somehow related to the rest of this PR?
I didn't see the team consensus at #818, so, surprising to see such unrelated change inside commits here...

Reply:
This probably could be another rebase problem.
https://github.com/node-fetch/node-fetch/blob/bea4a7cb732c54965544bc53324a9bb90fd1004e/src/request.js#L224-L227